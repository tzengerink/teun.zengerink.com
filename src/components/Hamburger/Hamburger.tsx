import classNames from 'classnames'

interface HamburgerProps {
  isOpen: boolean
  onClick: () => void
}

const Hamburger: React.FC<HamburgerProps> = ({ isOpen, onClick }) => {
  return (
    <div
      data-testid="hamburger"
      className={classNames('fixed', 'block', 'z-40', 'top-md', 'right-md', 'cursor-pointer')}
      onClick={onClick}
    >
      {['top', 'middle', 'bottom'].map((position) => {
        return (
          <span
            key={position}
            className={classNames(
              'block',
              'w-[24px]',
              'h-[3px]',
              'mx-auto',
              'my-[4px]',
              'bg-grey',
              'ease-linear',
              'duration-100',
              { 'translate-y-[8px]': isOpen && position === 'top', 'rotate-45': isOpen && position === 'top' },
              { 'opacity-0': isOpen && position === 'middle' },
              {
                'translate-y-[-6px]': isOpen && position === 'bottom',
                'rotate-[-45deg]': isOpen && position === 'bottom',
              },
              'md:hidden',
            )}
          ></span>
        )
      })}
    </div>
  )
}

export default Hamburger
