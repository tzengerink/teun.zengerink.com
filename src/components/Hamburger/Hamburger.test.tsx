import { fireEvent, render } from '@testing-library/react'
import Hamburger from './Hamburger'

describe('Hamburger', () => {
  const onClick = jest.fn()
  const renderComponent = (isOpen: boolean) => render(<Hamburger isOpen={isOpen} onClick={onClick} />)

  beforeEach(jest.clearAllMocks)

  it('renders correctly when open', () => {
    const { container } = renderComponent(true)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly when closed', () => {
    const { container } = renderComponent(false)
    expect(container).toMatchSnapshot()
  })

  it('triggers on click handler', () => {
    const { getByTestId } = renderComponent(false)
    const hamburger = getByTestId('hamburger')
    expect(onClick).not.toHaveBeenCalled()
    fireEvent.click(hamburger)
    expect(onClick).toHaveBeenCalled()
  })
})
