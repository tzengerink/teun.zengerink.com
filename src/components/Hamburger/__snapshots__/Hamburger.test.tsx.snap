// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger renders correctly when closed 1`] = `
<div>
  <div
    class="fixed block z-40 top-md right-md cursor-pointer"
    data-testid="hamburger"
  >
    <span
      class="block w-[24px] h-[3px] mx-auto my-[4px] bg-grey ease-linear duration-100 md:hidden"
    />
    <span
      class="block w-[24px] h-[3px] mx-auto my-[4px] bg-grey ease-linear duration-100 md:hidden"
    />
    <span
      class="block w-[24px] h-[3px] mx-auto my-[4px] bg-grey ease-linear duration-100 md:hidden"
    />
  </div>
</div>
`;

exports[`Hamburger renders correctly when open 1`] = `
<div>
  <div
    class="fixed block z-40 top-md right-md cursor-pointer"
    data-testid="hamburger"
  >
    <span
      class="block w-[24px] h-[3px] mx-auto my-[4px] bg-grey ease-linear duration-100 translate-y-[8px] rotate-45 md:hidden"
    />
    <span
      class="block w-[24px] h-[3px] mx-auto my-[4px] bg-grey ease-linear duration-100 opacity-0 md:hidden"
    />
    <span
      class="block w-[24px] h-[3px] mx-auto my-[4px] bg-grey ease-linear duration-100 translate-y-[-6px] rotate-[-45deg] md:hidden"
    />
  </div>
</div>
`;
